app/assets/javascripts/discourse/app/routes/tag-show.js

```diff
@@ -125,24 +137,24 @@ export default DiscourseRoute.extend(FilterModeMixin, {
     );
     const controller = this.controllerFor("tag.show");
 
-    if (controller.get("model.id")) {
-      if (this.category) {
+    if (controller.tag?.id) {
+      if (controller.category) {
         return I18n.t("tagging.filters.with_category", {
           filter: filterText,
-          tag: controller.get("model.id"),
-          category: this.get("category.name"),
+          tag: controller.tag.id,
+          category: controller.category.name,
         });
       } else {
         return I18n.t("tagging.filters.without_category", {
           filter: filterText,
-          tag: controller.get("model.id"),
+          tag: controller.tag.id,
         });
       }
     } else {
-      if (this.category) {
+      if (controller.category) {
         return I18n.t("tagging.filters.untagged_with_category", {
           filter: filterText,
-          category: this.get("category.name"),
+          category: controller.category?.name,
```

Comment:
`if (controller.category) {` just confirmed it's present 🙂 

```suggestion
          category: controller.category.name,
```